ts/features/idpay/onboarding/utility/hooks.ts

```diff
@@ -0,0 +1,51 @@
+import { useRef, useState } from "react";
+import {
+  NativeScrollEvent,
+  ScrollView,
+  useWindowDimensions
+} from "react-native";
+
+export const useInitiativeDetailsScrolling = () => {
+  const scrollViewRef = useRef<ScrollView>(null);
+  const markdownIsLoadingRef = useRef<boolean>(true);
+  const screenHeight = useWindowDimensions().height;
+  const [requiresScrolling, setRequiresScrolling] = useState<boolean>(true);
+
+  const onScrollViewSizeChange = (_: number, height: number) => {
+    // this method is called multiple times during the loading of the markdown
+    if (!markdownIsLoadingRef.current) {
+      setRequiresScrolling(height >= screenHeight);
+    }
+  };
+  const scrollToEnd = () => {
+    if (!markdownIsLoadingRef.current) {
+      scrollViewRef.current?.scrollToEnd();
+      setRequiresScrolling(false);
+    }
+  };
+
+  const handleIsScrollEnd = ({
+    layoutMeasurement,
+    contentOffset,
+    contentSize
+  }: NativeScrollEvent) => {
+    const paddingToBottom = 20;
```

Comment:
```suggestion
  const handleIsScrollEnd = (event: NativeScrollEvent) => {
    const { layoutMeasurement, contentOffset, contentSize } = event;
    const paddingToBottom = 20;
```

 improves readability, imho